arangod/Cluster/CreateCollection.cpp

```diff
@@ -183,18 +183,20 @@ bool CreateCollection::first() {
 
       // Steal buffer for maintenance feature
       _feature.storeShardError(database, collection, shard, eb.steal());
-      
+
       _result.reset(TRI_ERROR_FAILED, error.str());
-      // FIXMEMAINTENANCE: notify here?
+      notify();
       return false;
     }
-    
+
   } catch (std::exception const& e) { // Guard failed?
     std::stringstream error;
     error << "action " << _description << " failed with exception " << e.what();
     LOG_TOPIC(WARN, Logger::MAINTENANCE) << error.str();
     _result.reset(TRI_ERROR_ARANGO_DATABASE_NOT_FOUND, error.str());
-    // FIXMEMAINTENANCE: notify here?
+    // Notify not necessary, since no shard error created.
```

Comment:
How does a javascript client learn that its attempt to create a new collection failed if this never produces posts an error to Current?

Reply:
See similar code path in CreateDatabase.  All fail paths still exit the function at the bottom two lines:  notify(); return false;   The code path here and there should be same.

Reply:
I agree with you. Max told me this morning, that a error here should only happen when there was a race and a MaintenanceWorker tries to create the shard before the other created the database. 

However, we could report under `Current/Collections/<database>/...`. This needs a decision. @neunhoef